transcom/mymove
# update FE to account for new dateOfContact field

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16348)

## Summary

The purpose of this PR is to show the new date of contact field in the service items details section alongside the `timeMilitary` field. Also updates how we render `First available dates` and `customer contact attempts` to account for the case where there may be more than 2 of each. 

Check out [this slack thread if you would like extra context](https://ustcdp3.slack.com/archives/CTNTFJSBA/p1686772959097629)

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Run `make server_test`
2. All tests should pass
3. Create a new customer `HHG` move and move it along the system to make it available to the prime
4. Login into to the Prime UI and access your newly created move
5. Click the `Add Service Item` button
6. Scroll down to the form and select `Destination SIT` from the `Service item type` dropdown
7. The `date of attempted contact` field should be present - select a date, complete the form, and submit
8. It should now be reflected in the `mto_service_item_customer_contacts` table if you check your DB
9. You can also access the move as a `TOO` and check the move task order tab to check the service items section 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
Service Item Section
<img width="964" alt="Screenshot 2023-06-20 at 3 33 21 PM" src="https://github.com/transcom/mymove/assets/111077497/7c40c18c-1611-4710-9bae-9010c9ca7bcc">
